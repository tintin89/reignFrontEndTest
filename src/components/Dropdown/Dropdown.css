.dropdown__container{
  margin-top: 4rem;
  width: 100%;
  max-width: 1100px;
  display: flex;
}
.dropdown{
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 5px 12px;
  width: 240px;  
  border-radius: 4px;
  border: 1px solid #2e2e2e;
  background-color: #fff;
  font-family: Roboto;
  font-size: 14px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.57;
  letter-spacing: normal;
  color: #343434;
}

.dropdown__selected{
    display: flex;
    align-items: center;
}
.dropdown__selected span{
    margin-left: 0.3rem;
}

.downArrow{
  border: solid #2e2e2e;
  border-width: 0 2px 2px 0;
  display: inline-block;
  padding: 3px;
  transform: rotate(45deg);
  -webkit-transform: rotate(45deg);
}

.dropdownList{   
    z-index: 100;
    width: 263px;
    height: 139px;
    margin: 0 0 0 1px;    
    box-shadow: 0 2px 2px 0 #dad8d8;
    background-color: #fff;
    list-style: none;
    display: flex;
    flex-flow: column;
    position: absolute;
    transform: translateY(35px);    
}
.dropdownList li{  
    padding: 10px 10px; 
   cursor: pointer;
   height: 46px;
   display: flex;
   align-items: center;  
}
.dropdownList li:hover{
background-color: #eaeaea54;

}
.dropdownList li span{
   margin-left: 0.3rem;
   font-family: Roboto;
   font-size: 14px;
   font-weight: normal;
   font-stretch: normal;
   font-style: normal;
   line-height: 1.57;
   letter-spacing: normal;
   color: #343434;
}


@media (max-width: 600px){
  .dropdown__container{
    justify-content: center;
  }
}