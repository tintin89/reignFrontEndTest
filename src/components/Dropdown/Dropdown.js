import React,{useState,useEffect} from 'react';
import './Dropdown.css';
import react from '../../assets/react.png'
import angular from '../../assets/angular.png';
import vue from '../../assets/vue.png';
import { useSelector,useDispatch} from 'react-redux';
import {updateQuery} from '../../store/actions';

const mapState = (myStateApp) =>({
   query:myStateApp.query
})


const options = [{tag:"Angular",img:angular},{tag:"Reactjs",img:react},{tag:"Vuejs",img:vue}];

function Dropdown() {
    const dispatch = useDispatch();
    const {query} = useSelector(mapState);
    const [show,setShow] = useState(false);
    const [selected,setSelected] = useState({tag:"Select your news",img:""});


    useEffect(() => {       
       setSelected(query);
    }, [query])


    const handleSetShow = () =>{
        setShow(prevState=>!prevState)
    }

    const handleFilter = (e) =>{        
            setSelected(e);
            sessionStorage.setItem("query",JSON.stringify(e));
            dispatch(updateQuery(e));
            return handleSetShow();
      }

    return (
        <div className="dropdown__container">
            <div onClick={()=>handleSetShow()} className="dropdown">
                <div className="dropdown__selected">
                {selected?.img!=="" && <img alt={selected?.tag} src={selected?.img}/>}
                <span>{selected?.tag}</span>
                </div>
                <div className={`Arrow ${show && "upArrow"}`}></div>
            </div>
            <ul style={{display:show ? "flex" : "none"}} className="dropdownList">
              {
                  options.map((e,index)=>
                 selected?.tag!==e.tag && <li onClick={()=>handleFilter(e)} key={index}>
                   <img alt={e.tag} src={e.img}/>
                   <span>{e.tag}</span>
                  </li>
                  )
              }               
            </ul>
        </div>
    )
}

export default Dropdown
