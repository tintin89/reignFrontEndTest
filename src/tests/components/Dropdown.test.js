import React from 'react';
import  { mount}  from "enzyme";
import Dropdown from '../../components/Dropdown/Dropdown';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

describe('tests in  <Dropdown/>',()=>{   
    const initState = {
        query:"Reactjs",        
    };
    let store = mockStore(initState);

    const wrapper = mount(
        <Provider store={store}>
         <Dropdown/>
        </Provider>
    );

    test('Must show correctly the component',()=>{       
        
        expect(wrapper).toMatchSnapshot()
    }) 
      
})