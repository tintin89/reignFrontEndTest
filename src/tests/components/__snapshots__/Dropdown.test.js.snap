// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in  <Dropdown/> Must show correctly the component 1`] = `
<div
  className="dropdown__container"
>
  <div
    className="dropdown"
    onClick={[Function]}
  >
    <div
      className="dropdown__selected"
    >
      <img />
      <span />
    </div>
    <div
      className="Arrow false"
    />
  </div>
  <ul
    className="dropdownList"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <li
      key="0"
      onClick={[Function]}
    >
      <img
        alt="Angular"
        src="angular.png"
      />
      <span>
        Angular
      </span>
    </li>
    <li
      key="1"
      onClick={[Function]}
    >
      <img
        alt="Reactjs"
        src="react.png"
      />
      <span>
        Reactjs
      </span>
    </li>
    <li
      key="2"
      onClick={[Function]}
    >
      <img
        alt="Vuejs"
        src="vue.png"
      />
      <span>
        Vuejs
      </span>
    </li>
  </ul>
</div>
`;
